ablySpec/Auth.swift

```diff
@@ -529,5 +529,256 @@ class Auth : QuickSpec {
                 })
             }
         }
+
+        // RSA10
+        describe("authorise") {
+
+            // RSA10a
+            it("should create a token if needed and use it") {
+                let options = AblyTests.commonAppSetup()
+                var validToken: String?
+
+                // Create a new token
+                waitUntil(timeout: testTimeout) { done in
+                    let rest = ARTRest(options: options)
+
+                    rest.auth.authorise(nil, options: nil, callback: { tokenDetails, error in
+                        expect(rest.auth.method).to(equal(ARTAuthMethod.Token))
+                        expect(tokenDetails).toNot(beNil())
+                        expect(tokenDetails?.token).toNot(beEmpty())
+
+                        if let token = tokenDetails?.token {
```

Comment:
@mattheworiordan Because without it I need to force the unbox with `!` and it could raise an exception. I can also use the nil coalescing opertor `??` something like `validToken = tokenDetails?.token ?? ""` where if the `tokenDetails` is nil then it assigns an empty string.